homeassistant/components/tado/climate.py

```diff
@@ -401,9 +401,9 @@ def target_temperature(self) -> float | None:
 
     def set_timer(
         self,
-        temperature: float,
-        time_period: int,
-        requested_overlay: str,
+        temperature: float | None = None,
```

Comment:
The `temperature` parameter isn't optional and must be a float like value according to thr service schema.

Reply:
Ah, good catch. Addressed this in https://github.com/home-assistant/core/pull/113872

I've only reverted the previous change, but now that you mention it, it makes sense temperature is a required parameter.